android/src/com/frostwire/android/gui/fragments/BrowsePeerFragment.java

```diff
@@ -925,7 +924,7 @@ public void updateViews() {
 
         @Override
         public void enterState() {
-
+            updateHeader();
```

Comment:
so yes, I believe this commit demonstrates my idea.
You should bother doing the finger update only when in normal state, and perhaps you can keep calling the `updateHeader()` function without the parameter, and only use the explicit version `updateHeader(boolean)` in special cases.

so, something like

```java
private void updateHeader() {
    updateHeader(currentFragmentState == normalFragmentState);
}
```